src/cgmemmgr.cpp

```diff
@@ -856,8 +856,11 @@ uint8_t *RTDyldMemoryManagerJL::allocateCodeSection(uintptr_t Size,
                                                     StringRef SectionName)
 {
     // allocating more than one code section can confuse libunwind.
+#if defined(_COMPILER_MSAN_ENABLED_)
+    // TODO: Figure out why msan needs this.
     assert(!code_allocated);
```

Comment:
Why is this assert now only used when MSAN is present? The memory manager can't handle recursive allocations anyways, so the assert guards against that?

Reply:
Sorry, this was a typo. Obviously the condition should be inverted.

Reply:
Even so, I'm surprised that simply disabling the assert works for MSAN at all. Whenever I've disabled that assert the memory manager dies later on with a deeper error.

Reply:
I don't know - I haven't looked into it further. You're welcome to it ;)